src/storage/SimpleResourceStore.ts

```diff
@@ -106,66 +105,36 @@ export class SimpleResourceStore implements ResourceStore {
   }
 
   /**
-   * Converts the Representation to an array of Quads.
-   * @param representation - Incoming Representation.
-   * @throws {@link UnsupportedMediaTypeHttpError}
-   * If the representation is not a Quad stream.
+   * Copies the Representation by draining the original data stream and creating a new one.
    *
-   * @returns Promise of array of Quads pulled from the stream.
+   * @param data - Incoming Representation.
    */
-  private async parseRepresentation(representation: Representation): Promise<Quad[]> {
-    if (representation.dataType !== DATA_TYPE_QUAD) {
-      throw new UnsupportedMediaTypeHttpError('SimpleResourceStore only supports quad representations.');
-    }
-    return arrayifyStream(representation.data);
+  private async copyRepresentation(data: Representation): Promise<Representation> {
```

Comment:
```suggestion
  private async copyRepresentation(source: Representation): Promise<Representation> {
```

perhaps? (nit of course)